benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp/time_step_omp_gpu.cpp

```diff
@@ -0,0 +1,156 @@
+#include <iostream>
+#include <chrono>
+#include <vector>
+#include <cstdlib>
+
+// Kernels
+#include "../../kernels/c_family/continuity_kern.c"
+#include "../../kernels/c_family/momentum_u_kern.c"
+#include "../../kernels/c_family/momentum_v_kern.c"
+#include "../../kernels/c_family/boundary_conditions_kern.c"
+#include "../../kernels/c_family/time_update_kern.c"
+
+extern "C" void c_invoke_time_step(
+        // Fields
+        double * ssha_t,
+        double * sshn_t,
+        double * sshn_u,
+        double * sshn_v,
+        double * hu,
+        double * hv,
+        double * un,
+        double * vn,
+        double * ua,
+        double * ht,
+        double * ssha_u,
+        double * va,
+        double * ssha_v,
+        // Grid
+        int * tmask,
+        double * area_t,
+        double * area_u,
+        double * area_v,
+        double * dx_u,
+        double * dx_v,
+        double * dx_t,
+        double * dy_u,
+        double * dy_v,
+        double * dy_t,
+        double * gphiu,
+        double * gphiv,
+        // Scalars
+        int istep,
+        int internal_xstart,
+        int internal_xstop,
+        int internal_ystart,
+        int internal_ystop,
+        int width,
+        double rdt,
+        double cbfr,
+        double visc,
+        double omega,
+        double d2r,
+        double g
+  	){
+
+#pragma omp target enter data map (to: area_v[0:width*width], ssha_u[0:width*width], sshn_t[0:width*width], sshn_u[0:width*width], sshn_v[0:width*width], ssha_t[0:width*width], hu[0:width*width], hv[0:width*width], un[0:width*width], vn[0:width*width], area_t[0:width*width], ht[0:width*width], tmask[0:width*width], gphiu[0:width*width], dx_u[0:width*width], dx_v[0:width*width], dx_t[0:width*width], dy_u[0:width*width], dy_t[0:width*width], ua[0:width*width], va[0:width*width], area_u[0:width*width], ssha_v[0:width*width], dy_v[0:width*width], gphiv[0:width*width])
```

Comment:
Is width*width correct? Doesn't that assume a square domain?

Reply:
@rupertford I've now changed this to width*height and updated time_step_mod.f90 accordingly.